webpack.config.js

```diff
@@ -0,0 +1,82 @@
+const webpack = require("webpack");
+const path = require("path");
+const CopyWebpackPlugin = require("copy-webpack-plugin");
+const ExtractTextPlugin = require("extract-text-webpack-plugin");
+const pkg = require("./package");
+const widgetName = pkg.widgetName;
+const name = pkg.widgetName.toLowerCase();
+
+const widgetConfig = {
+    entry: `./src/components/${widgetName}Container.ts`,
+    output: {
+        path: path.resolve(__dirname, "dist/tmp"),
+        filename: `src/com/mendix/widget/custom/${name}/${widgetName}.js`,
+        libraryTarget: "umd"
+    },
+    resolve: {
+        extensions: [ ".ts", ".js" ],
+        alias: {
+            "tests": path.resolve(__dirname, "./tests")
+        }
+    },
+    module: {
+        rules: [
+            { test: /\.ts$/, loader: "ts-loader" },
+            { test: /\.css$/, loader: ExtractTextPlugin.extract({
+                fallback: "style-loader",
+                use: "css-loader"
+            }) },
+            { test: /\.scss$/, loader: ExtractTextPlugin.extract({
+                fallback: "style-loader",
+                use: [
+                    { loader: "css-loader" },
+                    { loader: "sass-loader" }
+                ]
+            }) }
+        ]
+    },
+    mode: "development",
+    devtool: "source-map",
+    externals: [ "react", "react-dom" ],
+    plugins: [
+        new CopyWebpackPlugin([
+                { from: "src/**/*.xml" },
+                { from: "src/**/*.png", to: `src/com/mendix/widget/custom/${name}/` }
+                ], {
+                    copyUnmodified: true
+                }),
+        new ExtractTextPlugin({ filename: `./src/com/mendix/widget/custom/${name}/ui/${widgetName}.css` }),
+        new webpack.LoaderOptionsPlugin({ debug: true })
```

Comment:
not sure if we need this option any longer for webpack4, let me do some research on it

Reply:
Not sure too